src/stepper/stepper.ts

```diff
@@ -1683,6 +1683,29 @@ function reduceMain(
       const [callee, args] = [node.callee, node.arguments]
       // source 0: discipline: any expression can be transformed into either literal, ident(builtin) or funexp
       // if functor can reduce, reduce functor
+
+      // Reduce all arguments until it is irreducible
+      for (let i = 0; i < args.length; i++) {
+        const currentArg = args[i]
+        if (!isIrreducible(currentArg, context)) {
+          paths[0].push('arguments[' + i + ']')
+          const [reducedCurrentArg, cont, path, str] = reduce(currentArg, context, paths)
+          const reducedArgs = [...args.slice(0, i), reducedCurrentArg, ...args.slice(i + 1)]
+          return [
+            ast.callExpression(callee as es.Expression, reducedArgs as es.Expression[], node.loc),
+            cont,
+            path,
+            str
+          ]
+        } else if (
+          currentArg.type === 'Identifier' &&
+          !(currentArg.name in context.runtime.environments[0].head)
+        ) {
+          throw new errors.UndefinedVariable(currentArg.name, currentArg)
+        }
+      }
+
+      //Reduce callee until it is irreducible
```

Comment:
Should we not reduce the callee, before the arguments?
Example:
```
((x => x)(x => x))(1 + 2);
```
currently evaluates the callee first, and then 1 + 2. That is the correct order.